src/bp/ui-admin/src/Pages/Bot/Details.jsx

```diff
@@ -187,13 +200,13 @@ class Bots extends Component {
     )
   }
 
-  handleInputChanged = event => this.setState({ [event.target.name]: event.target.value })
-  handleStatusChanged = status => this.setState({ status })
-  handleCategoryChanged = category => this.setState({ category })
+  handleInputChanged = event => this.setState({ [event.target.name]: event.target.value, formHasChanged: true })
```

Comment:
what if any of those fields are changed then reverted back to what it was, the `formHasChanged` property will still be true even if the values are the same as the original ones. Depending on the UX you wanted to provide, up to you to change your implementation